contents/blog/the-posthog-array-1-39-0.mdx

```diff
@@ -0,0 +1,112 @@
+---
+date: 2022-08-29
+title: 'Array 1.39.0: Moving person properties onto events'
+rootPage: /blog
+sidebar: Blog
+showTitle: true
+hideAnchor: true
+categories: ['Release notes', 'Product updates']
+featuredImage: ../images/blog/posthog-array-blog.png
+featuredImageType: full
+excerpt: PostHog 1.39.0 introduces a new beta option which could improve query speed by up to 400x. And that's just for starters!
+---
+
+Want to know more about what we're up to? [Subscribe to HogMail, our newsletter](/newsletter), which we send out every two weeks!
+
+> Running a self-hosted instance? Check out our [Upgrading PostHog guide](/docs/runbook/upgrading-posthog).
+
+## PostHog 1.39.0 release notes
+
+**Release highlights:**
+
+- Beta: [Person properties on events](#beta-person-properties-have-moved-on-to-events)
+- New: Paul to add a highlight here
+- New: Paul to add a highlight here
+- New: Paul to add a highlight here
+- Improved: Paul to add a highlight here
+- Improved: Paul to add a highlight here
+- Improved: Paul to add a highlight here
+- Apps: JOE to add
+
+### Beta: Person properties on events
+
+We're getting ready to make a substantial change to the way [persons](/manual/persons) and [events](/manual/events) work by essentially combining them and moving all persons properties _onto_ events. 1.39.0 gives you the chance to try the feature ahead of the full release by toggling your `actors_on_events_teams` [instance setting](/docs/self-host/configure/instance-settings).
+
+So, why have we moved person properties onto events? In part it's because it streamlines how we store data as we continue to scale, but it's also to unlock significant speed increases on queries involving both persons and events. Previously, PostHog stored personIDs and event information in separate tables, but this update simplifie
```

Comment:
There were two reasons actually: 
1. scale
2. being able to query based on person properties at the event time (feel free to list this one as the first item)

sth like 

So, why are we making this change? There's two reasons: 
1. By talking to customers we realised how they want to query data and look at insights, but it wasn't possible in PostHog in the past - specifically it wasn't possible to query based on person properties at the time of the event. Now that we're putting person properties on the events this will be the default. Note that it's still possible to filter insights based on latest properties too - that can be done by using cohorts.
2. Scale - previously we needed to merge events distinct id and persons tables now we can just look up all the info in the events table itself and indexes and materialising columns helps us speed up queries even more.